ttnn/CMakeLists.txt

```diff
@@ -412,7 +505,39 @@ if(WITH_PYTHON_BINDINGS)
     list(APPEND TTNN_PUBLIC_LINK_LIBRARIES pch_pybinds ${Python3_LIBRARIES})
 endif()
 
-add_library(ttnn SHARED ${TTNN_FINAL_SRC})
+set(ALL_TTNN_SUBLIBRARIES)
+
+function(add_ttnn_sublibrary SUBLIBRARY_NAME)
+    message("Adding sublibrary with ${SUBLIBRARY_NAME} ")
+    add_library(${SUBLIBRARY_NAME} OBJECT ${ARGN})
+    target_include_directories(${SUBLIBRARY_NAME} PUBLIC ${TTNN_PUBLIC_INCLUDE_DIRS})
+    target_link_libraries(${SUBLIBRARY_NAME} PUBLIC ${TTNN_PUBLIC_LINK_LIBRARIES})
+    target_link_directories(${SUBLIBRARY_NAME} PUBLIC ${TTNN_PUBLIC_LINK_DIRS})
+    target_precompile_headers(${SUBLIBRARY_NAME} PRIVATE ${TTNN_PRECOMPILED_HEADERS})
+    target_compile_options(${SUBLIBRARY_NAME} PUBLIC -MP -Wno-int-to-pointer-cast -fno-var-tracking)
+    if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
+        target_compile_definitions(${SUBLIBRARY_NAME} PUBLIC DISABLE_NAMESPACE_STATIC_ASSERT)
+    endif()
```

Comment:
can you explain this please?

Reply:
`ttnn` used this [line](https://github.com/tenstorrent/tt-metal/blob/fb408fb2081f8603f425781fc3a4123c7bdfd016/ttnn/CMakeLists.txt#L593), and I didn't change it for the sub libraries.